src/main/java/code401challenges/linkedlist/LinkedList.java

```diff
@@ -56,4 +51,67 @@ public String toString(){
         }
         return allValues.toString();
     }
+
+
+
+    public void append(int value){
+
+    }
+
+
+
+    public void insertBefore(int value,int newVal){
+//        check to make sure the list has stuff in it
+        if(head == null || head.next == null){
+            System.out.println("Sad eagle screams in sorrow :( the action is not possible.");

```

Comment:
Why the check on (head.next == null)? If there's one node in the list, and has the value you want to insert before, you should be allowed to do that.